keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,352 @@
+  Kubernetes cluster
+* **Cluster Default PodNetwork** - This is the initial cluster-wide PodNetwork
+  provided during cluster creation that is available to the Pod when no
+  additional networking configuration is provided in Pod spec.
+* **Primary PodNetwork** - This is the PodNetwork inside the Pod which interface
+  is used for the default gateway.
+ 
+
+### User Stories
+
+All user stories represent the type of use cases the multi-networking API should
+be able to support. References to technologies or exact products does not
+indicate that this API will directly support them. The set of requirements,
+defined out of these use cases, are the final indicator of what will be covered
+by this API and effort.
+
+#### Story #1
+As a Cloud Native Network Function (CNF) vendor my workloads deals with 2 types
+of network traffic: control plane and dataplane. For regulatory compliance I have
+to isolate these 2 traffic types. To achieve this I require 2 separate network
+interfaces inside my Pod. The underlying implementation (outside the Pod) has to
+ensure isolation on either Layer-2 or Layer-3.
+
+<p align="center">
+  <img src="mn-story-1.png?raw=true" alt="multi-network story 1 network L2 isolation"/>
+</p>
+
+#### Story #2
+As a Cloud Native Network Function (CNF) vendor I require a HW-based interface
+(e.g. SRIOV VF) to be provisioned to my workload Pod. I need to leverage that HW
+for performance purposes (high bandwidth, low latency), that my user-space
+application (e.g. DPDK-based) can use. The VF will not use the standard netdev
+kernel module. The Pod’s scheduling to the nodes should be based on hardware
+availability (e.g. devicePlugin or some other way).
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster operator I wish to isolate workloads based on
```

Comment:
I think this is an important point which needs to be called out more -- which is the role-based ACLs for network configuration that /may/ be independent of people who run the workloads.

Reply:
Not sure I fully understand, but we do call out "Personas" above and how those relate to PodNetwork handling.